core/src/trezor/ui/layouts/eckhart/__init__.py

```diff
@@ -1048,6 +1060,160 @@ async def confirm_ethereum_approve(
             TR.confirm_total__title_fee,
         )
 
+    async def confirm_trade(
+        title: str,
+        subtitle: str,
+        sell_amount: str,
+        buy_amount: str,
+        address: str,
+        account: str,
+        account_path: str,
+        token_address: str,
+        back_button: bool,
+    ) -> ui.UiResult:
+        from trezor.ui.layouts.menu import Menu, confirm_with_menu
+
+        trade_layout = trezorui_api.confirm_trade(
+            title=title,
+            subtitle=subtitle,
+            sell_amount=sell_amount,
+            buy_amount=buy_amount,
+            back_button=back_button,
+        )
+
+        menu = Menu.root(
+            [
+                create_details(
+                    TR.address__title_receive_address,
+                    [
+                        ("", address),
+                        (TR.words__account, account),
+                        (TR.address_details__derivation_path, account_path),
+                    ],
+                ),
+                create_details(TR.ethereum__token_contract, token_address),
+            ],
+            TR.send__cancel_sign,
+        )
+
+        return await confirm_with_menu(trade_layout, menu, "confirm_trade")
+
+    async def confirm_ethereum_payment_request(
+        recipient_name: str,
+        recipient: str,
+        texts: Iterable[str],
+        refunds: Iterable[tuple[str, str, str]],
+        trades: List[tuple[str, str, str, str, str]],
+        account: str | None,
+        account_path: str | None,
+        chain_id: str,
+        maximum_fee: str,
+        fee_info_items: Iterable[tuple[str, str]],
+        token_address: str,
+    ) -> None:
+        from trezor.ui.layouts.menu import Menu, confirm_with_menu
+
+        for text in texts:
+            await raise_if_cancelled(
+                trezorui_api.confirm_value(
+                    title=TR.words__confirm, value=text, description=None
```

Comment:
Nit:
- Having a consistent title would be preferable, but it's just a feeling.
- The screen doesn't have an importance for the confirm button to have the green gradient, IMO. Setting explicit verb parameter `buttons__confirm` would make the gradient grey.

```suggestion
                    title=TR.words__swap, value=text, description=None, verb=TR.buttons__confirm
```

Reply:
Agree with the button and changed 75cebd9e18d404d42097289e783966e74cc8b09f

Disagree with the title. Remember that this is SLIP-24, not "swaps", which is a subset of SLIP-24. Although not part of the design, it is part of the standard. The `CoinPurchaseMemo` could be entirely missing and the transaction would just confirm some text...